the-collab-lab/tcl-33-smart-shopping-list
# Delete Items from the Grocery List with Confirmation Prompt

## Description

This code enables users to remove grocery items from their list. To avoid erroneous deletions, the app provides users with a prompt to confirm the removal of the item by clicking a "yes" or "no" button. A "yes" click removes the item and a "no" click keeps the item on the grocery list. Although we considered moving the onClick functionality into the separate Prompt component, we decided that it made more sense for it to remain on the ListView page for clarity and ease of future refactoring.

## Related Issue

Closes #11 

## Acceptance Criteria

- User is able to delete an item from the shopping list
- Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
- Deletion should cause the associated record(s) in the database to be deleted

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before
![GroceryListNoDelete](https://user-images.githubusercontent.com/50176611/142261569-5f326608-67d3-4fa6-b3f4-7926198827cd.jpg)

### After
![GroceryListWithHazelnuts](https://user-images.githubusercontent.com/50176611/142261618-3422e45b-7e4e-4104-a096-485926b70f03.png)
![GroceryListDeletePrompt](https://user-images.githubusercontent.com/50176611/142261630-15156d8b-ff14-4d7f-ae00-8de7cb3b104d.png)
![GroceryListWithoutHazelnuts](https://user-images.githubusercontent.com/50176611/142261639-0a893d91-0d5c-4613-bdcc-8544aa21cd7e.png)

![DeleteItemGif](https://user-images.githubusercontent.com/50176611/142263546-1a8b6f25-2e59-4ba3-aa21-ee92e1202419.gif)

## Testing Steps / QA Criteria



- Make sure you are on the main branch of tcl-33-smart-shopping-list
- Git checkout jt-ms-delete_item to take you to our branch
- npm start
- The app should open in your browser
- If you have a token saved in your local storage, you should see the View List page with grocery items
- If you do not have a token saved, start a new list, add a couple of items, and go to the View List page to see your list
- The View List page should feature a list of items with checkboxes on the left and Delete buttons on the right
- Click on a Delete button
  - A prompt should appear at the top of the list asking if you want to delete the item
  - Click Yes if you want to delete the item. Both the grocery item and the prompt should disappear
  - Click No if you do not want to delete the item. The grocery list should remain complete and the prompt should disappear
